File: Week1/remove-duplicates-from-sorted-list-ii/tangxinye_solution1.py
class Solution:
    # @param head, a ListNode
    # @return a ListNode
     def deleteDuplicates(self, head):
        if head == None or head.next == None:
            return head
        helper = ListNode(0)
        helper.next = head
        pre = helper
        now = helper.next
        while pre.next:
            while now.next and now.next.val == pre.next.val:
                now = now.next
            if now == pre.next:
                pre = pre.next
                now = pre.next
            else:
                pre.next = now.next
        return helper.next
        

File: Week1/remove-duplicates-from-sorted-list-ii/tangxinye_solution1.java
public class Solution {
    public ListNode deleteDuplicates(ListNode head) {
       if(head==null||head.next==null)
        	return head;
        ListNode helper = new ListNode(0);
        helper.next=head;
        ListNode pre= helper;
        ListNode now= head;
        while(now!=null){
        	while(now.next!=null&&pre.next.val==now.next.val){
        		now=now.next;
        	}
        	if(pre.next!=now){
        		pre.next=now.next;
        	}else{
        		pre=pre.next;
        	}
        	now=now.next;
        }
        return helper.next;
    }
}
